src/util/loadDatasetGraph.ts

```diff
@@ -85,11 +99,77 @@ function createDatasetGraph(rawGraph: any): DatasetGraph {
     for (const term of ontology.values()) {
       term.in_use = !!term.n_cells || [...term.descendants].some((id) => ontology.get(id)!.n_cells > 0);
     }
+    const rootKeys = [...ontology].filter(([_,v])=>v.parents?.length===0).map(([k,_])=>k);    
+    rootKeys.forEach((key)=>{
+      const rootTerm = ontology.get(key);  
+      if (rootTerm) {
+        calcDepthBFS(ontology, rootTerm, datasetGraph.depthMaps[prefix])
+        calcHeightDFS(ontology, rootTerm, datasetGraph.heightMaps[prefix])        
+      }
+    })    
   }
 
   return datasetGraph;
 }
 
+/**
+ * Calculate depth of a node in the ontology
+ *
+ * @param ontology
+ * @param term
+ * @param depthMap
+ */
+ function calcDepthBFS(ontology: Ontology, term: OntologyTerm, depthMap: Map<string,number>): void {
+  const stack = [...term.children];
+  const depthStack: number[] = new Array(stack.length);
+  depthStack.fill(1);
+  depthMap.set(term.id,0);
+  if (term.depth===-1) term.depth=0;
+  while ( stack.length > 0 ) {
+    const id = stack.pop();
+    const depth = depthStack.pop();
+    if (id && depth) {
+      const newTerm = ontology.get(id);
+      if (newTerm) {
+        if (newTerm.depth === -1) newTerm.depth = depth;
+        const newDepths = new Array(newTerm.children.length);
+        newDepths.fill(depth+1);
+        stack.push(...newTerm.children);        
+        depthStack.push(...newDepths);
+        if (!depthMap.has(newTerm.id)) depthMap.set(newTerm.id, depth)
+      }
+    }
+  }
+}
+
+/**
+ * Calculate height of a node in the ontology from the furthest leaf
+ *
+ * @param ontology
+ * @param term
+ * @param heightMap
+ */
+function calcHeightDFS(ontology: Ontology, term: OntologyTerm, heightMap: Map<string,number>): number {
+  const heights = [];
+  for (const child of term.children) {
+    const childTerm = ontology.get(child);
+    if (childTerm) {
+      heights.push(calcHeightDFS(ontology, childTerm, heightMap)+1);  
+    }
+  }
+  const val = heights.length > 0 ? Math.max(...heights) : 0;
```

Comment:
pick a more descriptive variable name for the future reader.   E.g., "maxHeight"?